packages/be-config/src/apis/configurationFileApi.ts

```diff
@@ -18,23 +18,32 @@ import {DecoratedFastify, Handler} from '@mia-platform/custom-plugin-lib'
 
 import {CONFIGURATION_NAME} from '../constants'
 import {aclExpressionEvaluator} from '../utils/aclExpressionEvaluator'
-import {readConfigurationFile} from '../utils/configurationManager'
+import {readJsonConfigurationFile, readRawFile} from '../utils/configurationManager'
 import {referencesReplacer} from '../utils/referencesReplacer'
 
-const retrieveConfigurationFile = async(instanceConfig: any, configurationName: string) => {
+const retrieveJsonConfiguration = async(instanceConfig: any, configurationName: string, userGroups: string[]) => {
   const configurationPath = `${instanceConfig.PLUGINS_CONFIGURATIONS_PATH}/${configurationName}.json`
-  return readConfigurationFile(configurationPath)
+  const configurationContent = await readJsonConfigurationFile(configurationPath)
+  const configurationContentFiltered = aclExpressionEvaluator(configurationContent, userGroups)
+  return referencesReplacer(configurationContentFiltered)
+}
+
+const retrieveRawConfiguration = async(instanceConfig: any, configurationName: string) => {
+  const configurationPath = `${instanceConfig.PLUGINS_CONFIGURATIONS_PATH}/${configurationName}`
+  return readRawFile(configurationPath)
 }
 
 export const configurationFileApiHandlerBuilder: (fastifyInstance: DecoratedFastify) => Handler<any> = (fastifyInstance) => {
+  const instanceConfig: any = fastifyInstance.config
   return async(request, reply) => {
-    const instanceConfig: any = fastifyInstance.config
     if (instanceConfig.PLUGINS_CONFIGURATIONS_PATH) {
-      const userGroups = request.getGroups()
       // @ts-ignore
-      const configurationContent = await retrieveConfigurationFile(instanceConfig, request.params[CONFIGURATION_NAME])
-      const configurationContentFiltered = aclExpressionEvaluator(configurationContent, userGroups)
-      reply.send(referencesReplacer(configurationContentFiltered))
+      const configurationName: string = request.params[CONFIGURATION_NAME]
+      const fileContent = await Promise.any([
```

Comment:
It should be better to have some configuration to set if the file is a json or not.  
With this `Promise.any`, the *retrieveRawConfiguration* function could be more quick than *retrieveJsonConfiguration*, and the returned file could be a raw configuration also if the content is a json

Reply:
I think that this could happen only if we have the same name for both json and raw configuration.
Otherwise it would not happen because the `retrieveJsonConfiguration` append the `.json` suffix to the configuration path.

Reply:
Why it is the suffix `.json` added? I don't know if it is a good approach, I think it's a little weak. It is not possible to make it clearer?

Reply:
The suffix `json` has been added to make it implicit when the `element-composer` request a configuration.
We can remove it from here and move inside the composer, but this could be a breaking change for this API.

Reply:
Oh, ok for the suffix atm. I'd rather avoid a breaking change.

So, currently, if I set a configuration file the backend will add the suffix `.json`, correct?

Reply:
Exactly